command/runner.go

```diff
@@ -70,7 +78,15 @@ func NewRunner(
 		opt(runner)
 	}
 
+	// set default provider if none set
 	var err error
+	if runner.provider == nil {
+		runner.provider, err = pulumiCommand.NewProvider(e.Ctx, runner.namer.ResourceName("provider", connName), &pulumiCommand.ProviderArgs{})
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
How is the proper provider configured, for instance aws vs kubernetes ?

Reply:
`pulumi.Command` defines `ProviderArgs` as an empty struct. This is essentially a dummy provider.
The user can also define their own `Provider` if needed.